app/src/main/java/com/kickstarter/viewmodels/ProjectCardHolderViewModel.java

```diff
@@ -0,0 +1,337 @@
+  }
+
+  interface Outputs {
+    Observable<String> backersCountText();
+    Observable<Boolean> backingViewGroupIsGone();
+    Observable<String> categoryNameText();
+    Observable<String> deadlineCountdownText();
+    Observable<Boolean> featuredViewGroupIsGone();
+    Observable<List<User>> friendsForNamepile();
+    Observable<String> friendAvatarUrl();
+    Observable<Boolean> imageIsInvisible();
+    Observable<Boolean> friendBackingViewIsHidden();
+    Observable<Boolean> fundingUnsuccessfulTextViewIsGone();
+    Observable<Boolean> metadataViewGroupIsGone();
+    Observable<Project> projectOutput();
+    Observable<Integer> percentageFunded();
+    Observable<Boolean> percentageFundedProgressBarIsGone();
+    Observable<String> percentageFundedText();
+    Observable<String> photoUrl();
+    Observable<String> blurbText();
+    Observable<String> nameText();
+    Observable<Project> notifyDelegateOfProjectClick();
+    Observable<Boolean> potdViewGroupIsGone();
+    Observable<DateTime> projectCanceledAt();
+    Observable<DateTime> projectFailedAt();
+    Observable<Boolean> projectStateViewGroupIsGone();
+    Observable<DateTime> projectSuccessfulAt();
+    Observable<DateTime> projectSuspendedAt();
+    Observable<String> rootCategoryNameForFeatured();
+    Observable<Boolean> setDefaultTopPadding();
+    Observable<Boolean> starredViewGroupIsGone();
+    Observable<Boolean> successfullyFundedTextViewIsGone();
+  }
+
+  final class ViewModel extends ActivityViewModel<ProjectCardViewHolder> implements Inputs, Outputs {
+
+    public ViewModel(final @NonNull Environment environment) {
+      super(environment);
+
+      this.projectOutput = this.project;
+
+      this.backersCountText = this.project
+        .map(Project::backersCount)
+        .map(NumberUtils::format);
+
+      this.backingViewGroupIsGone = this.project
+        .map(p -> metadataForProject(p) != Metadata.BACKING);
+
+      this.blurbText = this.project
+        .map(Project::blurb);
+
+      this.categoryNameText = this.project
+        .map(Project::category)
+        .map(Category::name)
+        .compose(coalesce(""));
+
+      this.deadlineCountdownText = this.project
+        .map(ProjectUtils::deadlineCountdownValue)
+        .map(NumberUtils::format);
+
+      this.featuredViewGroupIsGone = this.project
+        .map(p -> metadataForProject(p) != Metadata.CATEGORY_FEATURED);
+
+      this.friendAvatarUrl = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends)
+        .map(friends -> friends.get(0).avatar().small());
+
+      this.friendBackingViewIsHidden = this.project
+        .map(Project::isFriendBacking)
+        .map(BooleanUtils::negate);
+
+      this.friendsForNamepile = this.project
+        .filter(Project::isFriendBacking)
+        .map(Project::friends);
+
+      this.fundingUnsuccessfulTextViewIsGone = this.project
+        .map(p ->
+          !p.state().equals(Project.STATE_CANCELED) &&
```

Comment:
i think for style we like to put the conditionals in the front of new lines i.e.

```java
!p.state().equals(Project.STATE_CANCELED)
&& !p.state().equals(Project.STATE_FAILED)
&& !p.state().equals(Project.STATE_SUSPENDED));
```